libcloud/storage/drivers/s3.py

```diff
@@ -1015,23 +1015,50 @@ class S3SAEastStorageDriver(S3StorageDriver):
     ex_location_name = 'sa-east-1'
 
 
-class S3RGWOutscaleConnection(S3Connection):
+class S3RGWOutscaleConnectionAWS4(SignedAWSConnection, BaseS3Connection):
+    service_name = 's3'
+    version = API_VERSION
+
+    def __init__(self, user_id, key, secure=True, host=None, port=None,
+                 url=None, timeout=None, proxy_url=None, token=None,
+                 retry_delay=None, backoff=None):
+
+        super(S3RGWOutscaleConnectionAWS4, self).__init__(user_id, key,
+                                                          secure, host,
+                                                          port, url,
+                                                          timeout,
+                                                          proxy_url, token,
+                                                          retry_delay,
+                                                          backoff,
+                                                          4)  # force aws4
+
+
+class S3RGWOutscaleConnectionAWS2(S3Connection):
     pass
 
 
 class S3RGWOutscaleStorageDriver(S3StorageDriver):
 
     def __init__(self, key, secret=None, secure=True, host=None, port=None,
                  api_version=None, region=S3_RGW_OUTSCALE_DEFAULT_REGION,
-                 **kwargs):
+                 signature_version=DEFAULT_SIGNATURE_VERSION, **kwargs):
         if region not in S3_RGW_OUTSCALE_HOSTS_BY_REGION:
             raise LibcloudError('Unknown region (%s)' % (region), driver=self)
         self.name = 'OUTSCALE Ceph RGW S3 (%s)' % (region)
         self.ex_location_name = region
         self.region_name = region
-        self.connectionCls = S3RGWOutscaleConnection
-        self.connectionCls.host = S3_RGW_OUTSCALE_HOSTS_BY_REGION[region]
+        self.signature_version = signature_version
+        self.host = S3_RGW_OUTSCALE_HOSTS_BY_REGION[region]
         super(S3RGWOutscaleStorageDriver, self).__init__(key, secret,
                                                          secure, host, port,
                                                          api_version, region,
                                                          **kwargs)
+
+    def _ex_connection_class_kwargs(self):
```

Comment:
@jmunhoz Thanks for the update and sorry to bug you again, but this still won't work correctly.

You are still manipulating connectionCls inside the driver constructor. You should get rid of this code and as mentioned above, update connection class constructor to take `signature_version` argument and correctly set host inside the connection class constructor.

This `_ex_connection_class_kwargs` method doesn't actually do anything at this point, it doesn't throw because connection class declares `**kwargs`. So in short, connection class constructor accepts this argument, but doesn't do anything with it.

In any case, I can go ahead, merge this into master as is and add a (failing) test case so this can be fixed in the future.